modules/dnn/src/layers/einsum_layer.cpp

```diff
@@ -0,0 +1,1178 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+
+#include "../precomp.hpp"
+#include "layers_common.hpp"
+#include <opencv2/dnn/shape_utils.hpp>
+
+namespace cv
+{
+namespace dnn
+{
+
+static bool IsTransposeReshapeForEinsum(const std::vector<size_t>& perm,
+                                        std::vector<int> input_dims,
+                                        MatShape& new_shape) {
+    // As long as the dims with values > 1 stay in the same order, it's a reshape.
+    // Example: Shape=(1,1,1024,4096) -> perm=(2,0,3,1).
+    size_t last_permuted_axis = 0;
+    for (size_t i = 0; i < perm.size(); ++i) {
+        if (input_dims[perm[i]] == 1)
+        continue;
+        if (perm[i] < last_permuted_axis)
+        return false;
+        last_permuted_axis = perm[i];
+    }
+    new_shape.assign(input_dims.begin(), input_dims.end());
+    for (size_t i = 0; i < perm.size(); ++i) {
+        new_shape[i] = input_dims[perm[i]];
+    }
+    return true;
+}
+
+Mat batchwiseMatMul(
```

Comment:
Can we create an object of FullyConnected layer and use it for inference? For example, `Permute` layer inside [ShuffleChannel](https://github.com/opencv/opencv/blob/4.x/modules/dnn/src/layers/shuffle_channel_layer.cpp)